log/application.go

```diff
@@ -32,14 +32,56 @@ func NewApplication(config config.Config) *Application {
 	return &Application{
 		instance: instance,
 		Writer:   NewWriter(instance.WithContext(context.Background())),
+		config:   config,
 	}
 }
 
 func (r *Application) WithContext(ctx context.Context) log.Writer {
-	switch r.Writer.(type) {
-	case *Writer:
-		return NewWriter(r.instance.WithContext(ctx))
-	default:
+	return NewWriter(r.instance.WithContext(ctx))
+}
+
+func (r *Application) Channel(channel string) log.Writer {
+	if channel == "" || r.config == nil {
+		return r.Writer
+	}
+
+	instance := logrus.New()
+	instance.SetLevel(logrus.DebugLevel)
+
+	if err := registerHook(r.config, instance, channel); err != nil {
+		color.Red().Println("Init facades.Log error: " + err.Error())
+		return nil
+	}
+
+	return NewWriter(instance.WithContext(context.Background()))
+}
+
+func (r *Application) Stack(channels []string) log.Writer {
+	if r.config == nil {
 		return r.Writer
 	}
+
+	instance := logrus.New()
+	instance.SetLevel(logrus.DebugLevel)
+
+	for _, channel := range channels {
+		if channel == "" {
+			continue
+		}
+
+		channelPath := "logging.channels." + channel
+		driver := r.config.GetString(channelPath + ".driver")
+
+		if driver == log.StackDriver {
+			color.Red().Println("stack driver can't include self channel")
+			return nil
+		}
```

Comment:
We may don't need to add this limitation, `registerHook` will handle the logic.

Reply:
Naah registerHook won't handle it.

Reply:
The stack function doesn't add channels into the config? Do you mean  this?

<img width="878" alt="image" src="https://github.com/goravel/framework/assets/24771476/eab309cd-5145-4e3b-badb-2035f2c9eec3">

The method `facades.Log().Stack([]string{"stack", "single"})` is allowed, you seem to be missing something?

Reply:
Oh, sorry, I was mistaking the channel for the driver.